secret.go

```diff
@@ -25,3 +29,51 @@ type SecretService interface {
 	// DeleteSecret removes a single secret from the secret store.
 	DeleteSecret(ctx context.Context, orgID ID, ks ...string) error
 }
+
+// SecretField contains a key string, and value pointer.
+type SecretField struct {
```

Comment:
I think there's some value in keeping the concerns of the secret service as simple as possible, instead of adding marshaling/unmarshaling logic (which is more a concern of the http layer).

I'm also not sure if we need the `SecretField` struct. We shouldn't need to store secrets directly in this way to any struct written to bolt/etcd.